include/DeterministicNoiseSimulator.hpp

```diff
@@ -0,0 +1,120 @@
+#ifndef DDSIM_DETERMINISTICNOISESIMULATOR_HPP
+#define DDSIM_DETERMINISTICNOISESIMULATOR_HPP
+
+#include "QuantumComputation.hpp"
+#include "Simulator.hpp"
+
+#include <cstddef>
+#include <map>
+#include <memory>
+#include <random>
+#include <string>
+#include <vector>
+
+class DeterministicNoiseSimulator: public Simulator {
+public:
+    //    DeterministicNoiseSimulator(std::unique_ptr<qc::QuantumComputation>& qc, const unsigned int step_number, const double step_fidelity):
+    //        qc(qc), step_number(step_number), step_fidelity(step_fidelity) {
+    //        if (step_number == 0) {
+    //            throw std::invalid_argument("step_number has to be greater than zero");
+    //        }
+    //    }
+
+    DeterministicNoiseSimulator(std::unique_ptr<qc::QuantumComputation>& qc, unsigned long long seed):
+        Simulator(seed), qc(qc) {
+    }
+
+    DeterministicNoiseSimulator(std::unique_ptr<qc::QuantumComputation>& qc, const std::string& noise_effects, double noise_prob):
+        qc(qc) {
+        setNoiseEffects(noise_effects);
+        setAmplitudeDampingProbability(noise_prob);
+    }
+
+    void setAmplitudeDampingProbability(double cGateNoiseProbability) {
+        //The probability of amplitude damping (t1) often is double the probability , of phase flip, which is why I double it here
+        noise_probability                            = cGateNoiseProbability;
+        sqrt_amplitude_damping_probability           = {sqrt(noise_probability * 2), 0};
+        one_minus_sqrt_amplitude_damping_probability = {sqrt(1 - noise_probability * 2), 0};
+    }
+
+    std::map<std::string, std::size_t> Simulate([[maybe_unused]] unsigned int shots) override{};
```

Comment:
This proposal sounds very much like a stochastic simulation. The advantage of deterministic simulation is that it is exact. I would like to keep it that way.

Reply:
Yes, that would be more similar to stochastic simulation but would provide a handy way to compare both. Doesn't have to be done right away, but I think it would be usefule. Let's discuss this over the next coffee.